_posts/2020-03-02-ios-deploys-super-weird-totally-normal.markdown

```diff
@@ -0,0 +1,113 @@
+pulled down by devices. Even if _most_ users have automatic updates turned on, those updates are typically
+installed overnight. Consequently, there's quite a lag between when we deploy an iOS update and when users run the
+code. While it only takes about a week for 80% of our users to update to the latest version, there's a very long
+tail after that.
+
+![Graph of in-use vesrions of Artsy's app, illustrating both the quick adoption of new updates by most users and the long tail of old versions that are never updated](/images/2020-03-02-ios-deploys-super-weird-totally-normal/graph.png)
+
+iOS software is executed in an environment that we don't control, that we can't push updates to, and most
+importantly, that we can't roll back deploys on. If we ship a version of our app with a bug, but then ship an
+updated version with a fix right away, there is absolutely no guarantee that users will install that update.
+_Ever_. That means that _every_ deploy we make to our iOS software requires a lot of confidence in that code. In
+contrast, rolling back a web deploy is quite painless. If we ship a bug to the web, no problem: just roll back the
+deploy, fix the bug, and re-deploy with the fix.
+
+(Note: some readers might be wondering why we don't take advantage of over-the-air updates to our React Native
+JavaScript bundle. This is definitely possible, but our app is brownfield with some native code and some React
+Native code and we haven't yet built out the infrastructure for this. As more and more code shifts to React Native,
+we plan to investigate OTA updates to JavaScript bundles.)
+
+The second reason to deploy iOS software on a schedule, rather than with continuous delivery, depends on the App
+Store review process. This is another big difference that takes web engineers a while to get used to. Whenever we
+ship a version of our iOS app, we ship it to Apple for review, and then Apple ships it to our users. I'm not going
+to debate the utility of Apple as an intermediary – the fact is that they own the iOS platform and these are the
+rules they have chosen. If you want to ship iOS software, then you have to abide by Apple's rules.
+
+App Store review isn't exactly QA. I mean, if we were to ship an app update and it crashes upon launch, Apple isn't
+likely to approve that update. But if we have a small bug buried somewhere in the app, we can't expect Apple to
+find it. Apple is only testing for adherence to their
+[App Store Review Guidelines](https://developer.apple.com/app-store/review/guidelines/). Apple is looking for
+things like: is the app trying to steal user data? Is the app displaying objectionable material, given its age
+rating? Is the app's description and App Store metadata correct? That kind of stuff.
+
+So not only do iOS software developers need a lot of confidence in every deploy, but they also need to abide by
+Apple's guidelines.
+
+Alright. Let's revisit the situation from earlier.
```

Comment:
how about "Let's return to our case study" to make it super clear which situation we're talking about?

Reply:
Could also be a new section - "To deploy, or not to deploy..."? maybe too cheesy 😄